Sources/IO/Geometry/STLWriter/index.js

```diff
@@ -0,0 +1,213 @@
+  const strips = polyData.getStrips() ? polyData.getStrips().getData() : null;
+
+  const n = [];
+  let v1 = [];
+  let v2 = [];
+  let v3 = [];
+
+  // Strips
+  if (strips && strips.length > 0) {
+    throw new Error('Unsupported strips');
+  }
+
+  // Polys
+  for (let i = 0; i < polys.length; ) {
+    const pointNumber = polys[i++];
+
+    if (pointNumber) {
+      v1 = [
+        points[polys[i] * 3],
+        points[polys[i] * 3 + 1],
+        points[polys[i++] * 3 + 2],
+      ];
+      v2 = [
+        points[polys[i] * 3],
+        points[polys[i] * 3 + 1],
+        points[polys[i++] * 3 + 2],
+      ];
+      v3 = [
+        points[polys[i] * 3],
+        points[polys[i] * 3 + 1],
+        points[polys[i++] * 3 + 2],
+      ];
+      if (transform) {
+        vec3.transformMat4(v1, v1, transform);
+        vec3.transformMat4(v2, v2, transform);
+        vec3.transformMat4(v3, v3, transform);
+      }
+
+      vtkTriangle.computeNormal(v1, v2, v3, n);
+
+      writer.writeTriangle(v1, v2, v3, n);
+    }
+  }
+  writer.writeFooter(polyData);
+  return writer.getOutputData();
+}
+
+// ----------------------------------------------------------------------------
+// Static API
+// ----------------------------------------------------------------------------
+
+export const STATIC = {
+  writeSTL,
+};
+
+// ----------------------------------------------------------------------------
+// vtkSTLWriter methods
+// ----------------------------------------------------------------------------
+
+function vtkSTLWriter(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkSTLWriter');
+
+  publicAPI.requestData = (inData, outData) => {
+    const input = inData[0];
+    if (!input || input.getClassName() !== 'vtkPolyData') {
+      vtkErrorMacro('Invalid or missing input');
+      return;
+    }
+    outData[0] = writeSTL(input, model.format, model.transform);
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+const DEFAULT_VALUES = {
+  format: FormatTypes.BINARY,
+  transform: null,
+};
+
+// ----------------------------------------------------------------------------
+
+export function extend(publicAPI, model, initialValues = {}) {
+  Object.assign(model, DEFAULT_VALUES, initialValues);
+
+  // Make this a VTK object
+  macro.obj(publicAPI, model);
+
+  // Also make it an algorithm with one input and one output
+  macro.algo(publicAPI, model, 1, 1);
+
+  macro.setGet(publicAPI, model, ['format', 'transform']);
+
+  // Object specific methods
+  vtkSTLWriter(publicAPI, model);
+}
+
+// ----------------------------------------------------------------------------
+
+export const newInstance = macro.newInstance(extend, 'vtkSTLWriter');
+
+// ----------------------------------------------------------------------------
+
+export default { newInstance, extend };
```

Comment:
`export default { newInstance, extend, ...STATIC };`

Reply:
I am fixing this right now!